src/redux/features/batches/batchesAsyncThunks.ts

```diff
@@ -0,0 +1,68 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { getTodaysDateFormatted } from '@terraware/web-components/utils/date';
+import { Response } from 'src/services/HttpService';
+import strings from 'src/strings';
+import { Batch, CreateBatchRequestPayload, NurseryTransfer } from 'src/types/Batch';
+import { BatchData, BatchId, UpdateBatchRequestPayloadWithId } from 'src/services/NurseryBatchService';
+import AccessionService from 'src/services/AccessionService';
+import { NurseryBatchService } from 'src/services';
+
+export type SavableBatch = (CreateBatchRequestPayload | UpdateBatchRequestPayloadWithId) & Batch;
+
+export const requestSaveBatch = createAsyncThunk(
+  'batches/save',
+  async (request: { batch: SavableBatch; timezone: string }, { rejectWithValue }) => {
+    const { batch, timezone } = request;
+
+    let response: (Response & BatchData) | (Response & BatchId) | undefined;
+    let responseQuantities: Partial<Response> = { requestSucceeded: true, error: undefined };
+
+    if (batch.id === -1) {
```

Comment:
Should this be `if (batch && batch.id === -1)` or `if (batch?.id === -1)`?

Reply:
By this point we have already guaranteed that the batch exists, we've narrowed it to `SavableBatch`. In other words, you can't call this request without a batch